packages/teleport/src/AppLauncher/AppLauncher.tsx

```diff
@@ -14,25 +14,77 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React from 'react';
+import React, { useCallback, useEffect } from 'react';
+
+import { useParams } from 'react-router';
+
 import { Flex, Indicator } from 'design';
+
 import { AccessDenied } from 'design/CardError';
 
-import useAppLauncher from './useAppLauncher';
+import useAttempt from 'shared/hooks/useAttemptNext';
 
-export default function Container() {
-  const state = useAppLauncher();
-  return <AppLauncher {...state} />;
-}
+import { UrlLauncherParams } from 'teleport/config';
+import service from 'teleport/services/apps';
+
+export function AppLauncher() {
+  const { attempt, setAttempt } = useAttempt('processing');
+
+  const params = useParams<UrlLauncherParams>();
+
+  const createAppSession = useCallback(async (params: UrlLauncherParams) => {
+    try {
+      let fqdn = params.fqdn;
+      if (!fqdn) {
+        const app = await service.getAppFqdn(params);
+
+        fqdn = app.fqdn;
+      }
+
+      const port = location.port ? ':' + location.port : '';
+      const session = await service.createAppSession(params);
 
-export function AppLauncher(props: ReturnType<typeof useAppLauncher>) {
-  if (props.status === 'failed') {
-    return <AccessDenied message={props.statusText} />;
+      await fetch(`https://${fqdn}${port}/x-teleport-auth`, {
+        method: 'POST',
+        credentials: 'include',
+        headers: {
+          'X-Cookie-Value': session.cookieValue,
+          'X-Subject-Cookie-Value': session.subjectCookieValue,
+        },
+      });
+
+      window.location.replace(`https://${fqdn}${port}`);
+    } catch (err) {
```

Comment:
```suggestion
    } catch ((err: Error)) {
```

Reply:
It's never guaranteed what is being thrown is of type `Error` -
<img width="629" alt="image" src="https://user-images.githubusercontent.com/7922109/212167626-0c30ee65-7493-435f-9837-cdfcb4df52e0.png">

Reply:
ah ok, then should we do this (create a helper func perhaps): https://github.com/gravitational/webapps/blob/5458f66bccbf7e3df879e98e7ee8096bb0f7665b/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts#L188

Reply:
I don't love that approach as it's assuming if we don't get an `Error`, that the thing thrown can be converted to a string. If it's an object that's thrown, that would put `[object Object]` into the message.

I've updated this catch to check for an instance of an error and use the error message provided if so. If not, it'll just fallback to a generic "Something went wrong"